src/libs/utils.js

```diff
@@ -160,3 +160,16 @@ export const convertValue = _.curry((type, value) => {
       throw new Error('unknown type for convertValue')
   }
 })
+
+export const renderTypeText = iotype => {
+  if (_.has('primitive_type', iotype)) {
+    return iotype.primitive_type
+  }
+  if (_.has('optional_type', iotype)) {
+    return `${_.get('optional_type.primitive_type', iotype)} (optional)`
+  }
+  if (_.has('array_type', iotype)) {
+    return `Array[${renderTypeText(_.get('array_type', iotype))}]`
+  }
+  throw new Error(`iotype ${iotype} does not contain a valid type`)
```

Comment:
I'm not sure that this is the right way to handle the situation where there are no matching types

Reply:
So one easy heuristic... assuming we have a bug in this logic, would we prefer the page to crash, or to display some kind of "Unrecognized Type" value and either wait to be told about it, or separately set up some kind of reporting?

Reply:
TOL: Maybe we could say `Type not supported` until we support all types?